src/drivers/boards/common/board_crashdump.c

```diff
@@ -0,0 +1,172 @@
+#include <px4_config.h>
+#include <px4_tasks.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include <nuttx/board.h>
+
+#include "board_config.h"
+
+#include <systemlib/hardfault_log.h>
+
+#if defined(CONFIG_STM32_SAVE_CRASHDUMP)
+static void copy_reverse(stack_word_t *dest, stack_word_t *src, int size)
+{
+	while (size--) {
+		*dest++ = *src--;
+	}
+}
+#endif
+
+__EXPORT void board_crashdump(uintptr_t currentsp, FAR void *tcb, FAR const uint8_t *filename, int lineno)
+{
+	if (!hardfault_check_status) {
```

Comment:
We had FLASH resource issues some targets. So this was not in all the targets.  On some targets it was an attempt to hint to the compiler to optimize this code out if the module hard_fault (application) was not in the config/cmake file. 

I am not sure if we need it any more with gcc 5.4.1 better optimization.

Lets remove it is see if CI passes it.